bin/release.py

```diff
@@ -0,0 +1,125 @@
+    for definition in project_release_definitions.value:
+        if definition.name == pipeline_name:
+            pipeline_release_definitions.append(definition)
+
+    if pipeline_release_definitions is None or len(pipeline_release_definitions) == 0:
+        error_message = f"Pipeline named {pipeline_name} not found in project {project}"
+        raise Exception(error_message)
+
+    if len(pipeline_release_definitions) > 1:
+        error_message = (
+            f"More than 1 Pipeline named {pipeline_name} found in project {project}"
+        )
+        raise Exception(error_message)
+
+    return pipeline_release_definitions[0]
+
+
+def populate_release_metadata(project, pipeline_name, release_client):
+    """Returns release metadata with required information"""
+    release_definition = get_release_definition(project, pipeline_name, release_client)
+    release_metadata = {
+        "definitionId": release_definition.id,
+    }
+    return release_metadata
+
+
+def release_status_match(release, expected_status_list):
+    """Returns True if status of all the environments of the release match any of the expected statuses"""
+    # Each release can have one or more environments (stages).
+    release_env_statuses = [environment.status for environment in release.environments]
+    # return all(status in expected_status_list for status in release_env_statuses)
+    for status in release_env_statuses:
+        if status not in expected_status_list:
+            return False
+    return True
+
+
+def wait_for_release(release_client, project, release_id):
+    """Wait for the azure devops release to finish"""
+    release = release_client.get_release(project, release_id)
+
+    # polling interval is set in accordance with the rate limits specified here:
+    # https://learn.microsoft.com/en-us/azure/devops/integrate/concepts/rate-limits?view=azure-devops
+    polling_interval_secs = 30
+
+    # Wait until the release have a complete status.
+    while not release_status_match(release, COMPLETED_STATUSES):
+        time.sleep(polling_interval_secs)
+        release = release_client.get_release(project, release_id)
+    return release
+
+
+def create_and_wait_for_azure_devops_release(
+    organization, project, pipeline_name, ADO_PAT
+):
+    """Creates and waits for an azure devops release to be finished"""
+    connection = create_ado_connection(organization, ADO_PAT)
+    release_client = connection.clients.get_release_client()
+    release_metadata = populate_release_metadata(project, pipeline_name, release_client)
+
+    triggered_release = release_client.create_release(release_metadata, project)
+    release_url = f"https://dev.azure.com/{organization}/{project}/_releaseProgress?_a=release-pipeline-progress&releaseId={triggered_release.id}"
+    print(
+        f"Successfully triggered a release. Waiting for the release to be completed.\nMore details on the release can be found here: {release_url}"
+    )

```

Comment:
Strings placed on multiple lines _without_ a `,` in Python will be considered the same string, so you can make this a little more readable like so.

```suggestion
    print(
        "Successfully triggered a release. Waiting for the release to be completed.\n"
        f"More details on the release can be found here: {release_url}"
    )

```